scripts/provisioning/configcmd.py

```diff
@@ -208,6 +210,62 @@ def get_endpoint(self, confstore_key, expected_token,  endpoint_type):
       raise S3PROVError(f"{confstore_key} does not specify endpoint fqdn {endpoint} for endpoint type {endpoint_type}")
     return endpoint[expected_token]
 
+
+  def push_s3_ldap_schema(self):
+      """ Push s3 ldap schema with below checks,
+          1. While pushing schema, global lock created in consul kv store as <index, key, value>.
+             e.g. <s3_consul_index, component>s3>openldap_lock, machine_id>
+          2. Before pushing s3 schema,
+             a. Check for s3 openldap lock from consul kv store
+             b. if lock is None/machine-id, then go ahead and push s3 ldap schema.
+             c. if lock has other values except machine-id/None, then wait for the lock and retry again.
+          3. Once s3 schema is pushed, delete the s3 key from consul kv store.
+      """
+      ldap_lock = False
+      self.logger.info('checking for concurrent execution scenario for s3 ldap scheam push using consul kv lock.')
+      openldap_key=self.get_confkey("S3_CONSUL_OPENLDAP_KEY")
+      consul_index=self.get_confkey("S3_CONSUL_INDEX")
+      consul_endpoint_url=self.get_endpoint("CONFIG>CONFSTORE_CONSUL_ENDPOINTs", "fqdn", "tcp")
+      consul_endpoint_port=self.get_endpoint("CONFIG>CONFSTORE_CONSUL_ENDPOINTs", "port", "tcp")
+      # consul url will be : consul://consul-server.default.svc.cluster.local:8500
+      consul_url='consul://' + f'{consul_endpoint_url}' + ':' + f'{consul_endpoint_port}'
+      self.logger.info('loading consul service with index as:', f'{consul_index}', 'and consul endpoint URL as:', f'{consul_url}')
+      Conf.load(f'{consul_index}', f'{consul_url}')
+      while(True):
+          try:
+              opendldap_val = Conf.get(f'{consul_index}', f'{openldap_key}')
+              self.logger.info('openldap value is:', f'{opendldap_val}')
+              if opendldap_val is None:
+                  self.logger.info('Setting confstore value for key :', f'{openldap_key}', ' and value as :', f'{self.machine_id}')
+                  Conf.set(f'{consul_index}', f'{openldap_key}', f'{self.machine_id}')
+                  self.logger.info('Saving confstore with latest value')
+                  Conf.save(f'{consul_index}')
+                  time.sleep(5)
+                  continue
+              if opendldap_val == self.machine_id:
+                  self.logger.info('Found lock already acquired hence enabling flag')
+                  ldap_lock = True
+                  break
+              if opendldap_val != self.machine_id:
+                  self.logger.info('Ignoring given entry for this iteration')
+                  ldap_lock = False
+                  break
+
+          except Exception as e:
+              self.logger.error('Exception occured while connecting consul service endpoint', f'{e}')
+              break
+      if ldap_lock == True:
+        # push openldap schema
+        self.logger.info('Pushing s3 ldap schema ....!!')
+        self.configure_s3_schema()
```

Comment:
if nodes run this sequentially, how multiple self.configure_s3_schema() schema are handled. do we skip if schema is already pushed

Reply:
this will wait for the lock to obtain from consul. once lock is acquired, schema will be pushed from that node.

Reply:
@bkirunge7  ^^